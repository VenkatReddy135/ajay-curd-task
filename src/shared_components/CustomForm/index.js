import React from "react";
import PropTypes from "prop-types";
import CustomInput from "../../shared_components/CustomInput";
import CustomButton from "../../shared_components/CustomButton";

export default function CustomForm(props) {
  return (
    <div>
      <form className="CustomForm">
        <div className={`container border border-info p-4 mt-4`}>
          <h2>{props.formData?.employee_name} Details:</h2>
          <div className="row">
            <div className="col-4">
              <label>Employee Name:</label>
              <CustomInput
                inputType="text"
                inputValue={props.formData?.employee_name}
                inputClass={`${
                  props.mode === "view" ? "border-0" : ""
                } form-control inputWidth`}
              />
            </div>
            <div className="col-4">
              <label>Employee Age:</label>
              <CustomInput
                inputType="text"
                inputValue={props.formData?.employee_age}
                inputClass={`${
                  props.mode === "view" ? "border-0" : ""
                } form-control inputWidth`}
              />
               </div>
                <div className="col-4">
                <label>Employee Salary:</label>
                <CustomInput
                  inputType="text"
                  inputValue={props.formData?.employee_salary}
                  inputClass={`${
                    props.mode === "view" ? "border-0" : ""
                  } form-control inputWidth`}
                />
              </div>
           
            {/* <div className="row">
            
            </div> */}
            <div className="row">
              <CustomButton
                buttonClass={`button btn btn-secondary backButton`}
                handleClick={props.buttonAction}
                buttonValue={props.buttonText}
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  );
}
CustomForm.propTypes = {
  formData: PropTypes.any.isRequired,
  mode: PropTypes.string,
  buttonText: PropTypes.string,
  buttonAction: PropTypes.func.isRequired,
};

CustomForm.defaultProps = {
  formData: {},
  buttonAction: () => {},
};
