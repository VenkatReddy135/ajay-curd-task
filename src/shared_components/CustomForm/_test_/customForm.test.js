import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import CustomForm from "../index";
import { cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(<CustomForm />);
});
const props = {
  formData: {},
  buttonAction: () => {},
};
describe("Custom Component Testing", () => {
  it("renders component", () => {
    const tree = renderer.create(<CustomForm {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CustomForm {...props} />, div);
  });
  it("should have CustomForm class name", () => {
    console.log(wrapper);
    expect(wrapper.find(".CustomForm")).toHaveLength(1);
  });

  afterEach(() => {
    cleanup();
  });
});
