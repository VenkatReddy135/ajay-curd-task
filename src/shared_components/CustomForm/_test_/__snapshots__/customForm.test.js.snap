// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Component Testing renders component 1`] = `
<div>
  <form
    className="CustomForm"
  >
    <div
      className="container border border-info p-4 mt-4"
    >
      <h2>
         Details:
      </h2>
      <div
        className="row"
      >
        <div
          className="col-4"
        >
          <label>
            Employee Name:
          </label>
          <input
            className=" form-control inputWidth"
            placeholder=""
            type="text"
          />
        </div>
        <div
          className="col-4"
        >
          <label>
            Employee Age:
          </label>
          <input
            className=" form-control inputWidth"
            placeholder=""
            type="text"
          />
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <label>
              Employee Salary:
            </label>
            <input
              className=" form-control inputWidth"
              placeholder=""
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <button
            className="button btn btn-secondary backButton"
            onClick={[Function]}
          >
            
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
